src/shared/install.c

```diff
@@ -2271,15 +2271,15 @@ static int install_context_mark_for_removal(
                 if (r == -ENOLINK) {
                         log_debug_errno(r, "Name %s leads to a dangling symlink, removing name.", i->name);
                         r = install_changes_add(changes, n_changes, INSTALL_CHANGE_IS_DANGLING, i->path ?: i->name, NULL);
-                        if (r < 0)
+                        if (r == -ENOMEM)
                                 return r;
                 } else if (r == -ENOENT) {
                         if (i->auxiliary)  /* some unit specified in Also= or similar is missing */
                                 log_debug_errno(r, "Auxiliary unit of %s not found, removing name.", i->name);
                         else {
                                 log_debug_errno(r, "Unit %s not found, removing name.", i->name);
                                 r = install_changes_add(changes, n_changes, r, i->path ?: i->name, NULL);
-                                if (r < 0)
+                                if (r == -ENOMEM)
```

Comment:
so this one is the first one where the 3rd argument is r. And in this specific case r is actually -ENOENT, as we verified earlier, hence maybe change this failure condition here to `if (r != -ENOENT) return r;`